src/libclang/libclang_parser.cpp

```diff
@@ -243,11 +248,30 @@ cppast::libclang_compile_config::libclang_compile_config(
                 add_flag(std::move(flag));
             }
             else if (flag == "-std")
+            {
                 // standard
+                try
+                {
+                    // detect whether the language standard is a C standard
+                    use_c_ = is_c_standard(to_standard(args));
```

Comment:
`args` can also be `gnuXX`, which isn't handled by `to_standard`. This is another indicator that the function should be private.

Reply:
I simplified this logic just to check for the string `"++"` in the language standard. IMO there's no need to have a function mapping a string to the language standard if this is the only use of that.

Reply:
Yeah, that's elegant.